### ISSUE TITLE
Tuning Timeout for google geocoding

### ISSUE THREAD

**Issue by etiennepierrot** [chadly/Geocoding.net#36]
I see no way to customize the timeout of HTTP call to google. Maybe i miss something, if not, i can make a PR : 
* adding a property Timeout nullable in GoogleGeocoder class
* use this property only if not null in 

var req = WebRequest.Create(url) as HttpWebRequest;

**Comment by chadly** [comment]
Yes, that seems like a good implementation. I would be appreciative if you wanted to submit a PR.

**Comment by WizX20** [comment]
For now I use this to be sure it does not take 20 seconds to time out:
`// run geo-coding with timeout
            IEnumerable<GoogleAddress> addresses = null;
            try {
                var task = Task.Run(() => geocoder.Geocode(term));
                int timeout = int.Parse(WebConfigurationManager.AppSettings[Constants.AppSettings.WEBSITE_GOOGLE_GEOCODE_TIMEOUT]);
                if (task.Wait(TimeSpan.FromSeconds(timeout))) {
                    addresses = task.Result;
                } else {
                    addresses = null;
                    Log.Error("Google geocoder timed out (Website.Core.GoogleGeoCoding)!", "GoogleGeoCoding");
                }
            }
            catch (Exception ex) {
                Log.Error("Unable to geocode location term!", ex, "GeoCodingHelper");
            }`
